appknox/reports.go

```diff
@@ -68,3 +80,60 @@ func (s *ReportsService) List(ctx context.Context, fileID int) ([]*ReportResult,
 	// return drfResponse.Results, &resp, nil
 
 }
+
+//Get Signed URL to download Summary CSV report Data
+func (s *ReportsService) GetDownloadUrlCSV(ctx context.Context, reportID int) (string, error) {
+	url := fmt.Sprintf("/api/v2/reports/%d/summary_csv", reportID)
+	request, err := s.client.NewRequest("GET", url, nil)
+	var drfResponseReportDownloadUrl DRFResponseReportDownloadUrl
+	resp, err := s.client.Do(ctx, request, &drfResponseReportDownloadUrl)
+	if resp != nil && resp.StatusCode == 404 {
+		id := strconv.Itoa(reportID)
+		return "", errors.New("Report with ID " + id + " doesn't exist. Are you sure " + id + " is a reportID?")
+	}
+	return drfResponseReportDownloadUrl.Url, err
+
+}
+
+//Download Report Data from Url to buffer
+func (s *ReportsService) DownloadReportData(ctx context.Context, downloadUrl string) (*http.Response, error) {
+
+	request, err := s.client.NewRequest("GET", downloadUrl, nil)
+	resp, err := s.client.Reports.client.DoTxt(request)
```

Comment:
Make it generic so that it can download any type of report.

Reply:
This is a generic function that will download report from the signed download URL to the response body buffer. can be used any type of report where response is report data itself irrespective of format